src/index.ts

```diff
@@ -4,16 +4,29 @@ import { PluginFunction } from '@graphql-codegen/plugin-helpers';
 import { pascalCase } from 'pascal-case';
 import { upperCase } from 'upper-case';
 
-type EnumValuesTypes = 'upper-case#upperCase' | 'pascal-case#pascalCase';
+type NamingConvention = 'upper-case#upperCase' | 'pascal-case#pascalCase' | 'keep';
+
+const createNameConverter = (convention: NamingConvention) => (value: string) => {
+    switch (convention) {
+        case 'upper-case#upperCase':
+            return upperCase(value || '');
+        case 'pascal-case#pascalCase':
+            return pascalCase(value || '');
+        case 'keep':
+            return value;
+        default:
+            // default to pascal case in case of unknown values
+            return pascalCase(value || '');
```

Comment:
I did not check what happens if a something else than the supported values is specified in the config options.

Reply:
I guess that should use the default value.
You can maybe grop the `case 'pascal-case#pascalCase':` and `default:` together, but keep it like that if you prefer

Reply:
I have grouped the options and explicited the fallthrough with a comment as suggested in the [eslint doc](https://eslint.org/docs/rules/no-fallthrough).